### PR TITLE
Add copy goal

### PR DESCRIPTION
**Pull request by basvandenbrink**
Add `copy` goal for copying container images that are generated as part of the `build` mojo to a different registry. This is useful when integrating the `podman-maven-plugin` in the gitflow, using the gitflow-helper-maven-plugin. Also adds the generation of the `container-catalog.txt` file as part of the `build` mojo. This file is used during the copy for determining which images need to be copied.

Uses skopeo under the hood.

### CODE REVIEW

File: src/test/java/nl/lexemmens/podman/service/PodmanExecutorServiceTest.java [hunk 1]
```diff
@@ -52,7 +54,7 @@ public class PodmanExecutorServiceTest {
 
     @Before
     public void setup() {
-        initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (lexemmens) [inline src/test/java/nl/lexemmens/podman/service/PodmanExecutorServiceTest.java:57]: TIL :writing_hand:

File: src/main/java/nl/lexemmens/podman/CopyMojo.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        List<ArtifactRepository> remoteArtifactRepositories;
+        String sourceCatalogRepository = skopeo.getCopy().getSourceCatalogRepository();
+
+        if (sourceCatalogRepository == null) {
+            getLog().info("Using all remote repositories to find container catalog.");
+            remoteArtifactRepositories = project.getRemoteArtifactRepositories();
+        } else {
+            Optional<ArtifactRepository> repository = project.getRemoteArtifactRepositories()
+                    .stream()
+                    .filter(r -> r.getId().equals(sourceCatalogRepository))
```
Comment (lexemmens) [inline src/main/java/nl/lexemmens/podman/CopyMojo.java:131]: We could do with changing `r` to `repository`

**Review by lexemmens — APPROVED** [review]
No major issues spotted. Some classes could still use some Javadoc though.

File: src/main/java/nl/lexemmens/podman/CopyMojo.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        List<RemoteRepository> remoteRepositories = getRemoteRepositories();
+        List<String> cataloguedImages = getCatalog(tempSession, remoteRepositories);
+        Map<String, String> transformedImages = performTransformation(cataloguedImages);
+
+        for (Map.Entry<String, String> imageEntry : transformedImages.entrySet()) {
+            copyImage(hub, imageEntry.getKey(), imageEntry.getValue());
+        }
+
+        if (skopeo.getCopy().getDisableLocal()) {
+            if (tempRepo != null) {
```
Comment (lexemmens) [inline src/main/java/nl/lexemmens/podman/CopyMojo.java:111]: These if statements can be merged :-)

File: src/main/java/nl/lexemmens/podman/CopyMojo.java (hunk 2 above)
Comment (basvandenbrink) [reply src/main/java/nl/lexemmens/podman/CopyMojo.java:131]: Will change it to `repo`, since `repository` conflicts with the var defined above.

**Pull request merged**